modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/hash/RemoveHashIndexRowInvokeClosure.java

```diff
@@ -84,7 +84,9 @@ public void afterCompletion() throws IgniteInternalCheckedException {
         if (indexColumns.link() != NULL_LINK) {
             assert operationType == OperationType.REMOVE;
 
-            freeList.removeDataRowByLink(indexColumns.link());
+            if (freeList != null) {
+                freeList.removeDataRowByLink(indexColumns.link());
+            }
```

Comment:
How's this possible? If link exists, freeList must exist.

Reply:
Fix it.